beacon-chain/execution/engine_client.go

```diff
@@ -390,30 +390,31 @@ func (s *Service) ExecutionBlocksByHashes(ctx context.Context, hashes []common.H
 	numOfHashes := len(hashes)
 	elems := make([]gethRPC.BatchElem, 0, numOfHashes)
 	execBlks := make([]*pb.ExecutionBlock, 0, numOfHashes)
-	errs := make([]error, 0, numOfHashes)
+	errs := make([]*error, 0, numOfHashes)
```

Comment:
Could you add some commentary here for future readers? This seems like an odd pattern to someone that doesn't have the full context of how gethRPC writes errors.

Something like: 
```suggestion
	// errs must be a slice of pointers to prevent errors being overwritten. See PR#11960 for more context.
	errs := make([]*error, 0, numOfHashes)
```

Reply:
After looking at the code again I found that we could simplify this altogether by getting rid of the errs array and check for the error of the batched elements instead. This should improve readability without using odd patterns :)